java/client/src/main/java/glide/api/RedisClusterClient.java

```diff
@@ -1072,6 +1080,14 @@ public CompletableFuture<Object[]> scan(ClusterScanCursor cursor, ScanOptions op
                         result -> new Object[] {new NativeClusterScanCursor(result[0].toString()), result[1]});
     }
 
+    @Override
+    public CompletableFuture<Object[]> scanBinary(ClusterScanCursor cursor, ScanOptions options) {
```

Comment:
I afraid you need new class `ScanOptionsBinary`

Reply:
The Binary options are useless and don't provide any additional benefit.  They are phasing them out.

Reply:
There is `String matchPattern` unfortunately

Reply:
protobuf doesn't accept anything but strings, so we cannot sent binary scan options